dbms/src/Storages/KVStore/Read/LearnerRead.cpp

```diff
@@ -12,523 +12,52 @@
-        if (!can_stale_read)
-        {
-            if (auto ori_read_index = mvcc_query_info.getReadIndexRes(region_id); ori_read_index)
-            {
-                auto resp = kvrpcpb::ReadIndexResponse();
-                resp.set_read_index(ori_read_index);
-                batch_read_index_result.emplace(region_id, std::move(resp));
-            }
-            else
-            {
-                auto & region = regions_snapshot.find(region_id)->second;
-                batch_read_index_req.emplace_back(GenRegionReadIndexReq(*region, read_index_tso));
-            }
-        }
-        else
-        {
-            batch_read_index_result.emplace(region_id, kvrpcpb::ReadIndexResponse());
-            ++stats.num_stale_read;
-        }
-    }
-    return batch_read_index_req;
-}
-
-void doBatchReadIndex(
-    std::vector<kvrpcpb::ReadIndexRequest> & batch_read_index_req,
-    std::unordered_map<RegionID, kvrpcpb::ReadIndexResponse> & batch_read_index_result,
-    KVStorePtr & kvstore,
-    TMTContext & tmt)
-{
-    const auto & make_default_batch_read_index_result = [&](bool with_region_error) {
-        for (const auto & req : batch_read_index_req)
-        {
-            auto resp = kvrpcpb::ReadIndexResponse();
-            if (with_region_error)
-                resp.mutable_region_error()->mutable_region_not_found();
-            batch_read_index_result.emplace(req.context().region_id(), std::move(resp));
-        }
-    };
-    [&]() {
-        if (!tmt.checkRunning(std::memory_order_relaxed))
-        {
-            make_default_batch_read_index_result(true);
-            return;
-        }
-        kvstore->addReadIndexEvent(1);
-        SCOPE_EXIT({ kvstore->addReadIndexEvent(-1); });
-        if (!tmt.checkRunning())
-        {
-            make_default_batch_read_index_result(true);
-            return;
-        }
-
-        /// Blocking learner read. Note that learner read must be performed ahead of data read,
-        /// otherwise the desired index will be blocked by the lock of data read.
-        if (kvstore->getProxyHelper())
-        {
-            auto res = kvstore->batchReadIndex(batch_read_index_req, tmt.batchReadIndexTimeout());
-            for (auto && [resp, region_id] : res)
-            {
-                batch_read_index_result.emplace(region_id, std::move(resp));
-            }
-        }
-        else
-        {
-            // Only in mock test, `proxy_helper` will be `nullptr`. Set `read_index` to 0 and skip waiting.
-            make_default_batch_read_index_result(false);
-        }
-    }();
-}
-
-} // namespace
-
 LearnerReadSnapshot doLearnerRead(
-    const TiDB::TableID logical_table_id,
-    MvccQueryInfo & mvcc_query_info_,
+    const TableID logical_table_id,
+    MvccQueryInfo & mvcc_query_info,
     bool for_batch_cop,
     Context & context,
     const LoggerPtr & log)
 {
     assert(log != nullptr);
+    // disagg compute node should not execute learner read
```

Comment:
```suggestion
    // disagg compute node should not execute learner read
```

Reply:
I'd like to keep it, sometime I easily forget to pass a valid logger in unit tests